scripts/create_compdb.py

```diff
@@ -23,145 +23,48 @@
 SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 """
 
-import json
-import re
 import subprocess
-import sys
-from pathlib import Path
 
 import scripts_utils
 
-scripts_utils.chdir_repo_root()
-directory = Path.cwd()
-
-# We use the `BAZEL` environment variable if present. If not, then we try to
-# use `bazelisk` and then `bazel`.
-bazel = scripts_utils.locate_bazel()
-
-# Load compiler flags. We do this first in order to fail fast if not run from
-# the workspace root.
-print("Reading the arguments to use...")
-try:
-    with open("compile_flags.txt") as flag_file:
-        arguments = [line.strip() for line in flag_file]
-except FileNotFoundError:
-    sys.exit(Path(sys.argv[0]).name + " must be run from the project root")
-
-# Prepend the `clang` executable path to the arguments that looks into our
-# downloaded Clang toolchain.
-arguments = ["clang++"] + arguments
-
-print("Building compilation database...")
-
-# Find all of the C++ source files that we expect to compile cleanly as
-# stand-alone files. This is a bit simpler than scraping the actual compile
-# actions and allows us to directly index header-only libraries easily and
-# pro-actively index the specific headers in the project.
-source_files_query = subprocess.check_output(
-    [
-        bazel,
-        "query",
-        "--keep_going",
-        "--output=location",
-        # Workaround for https://github.com/bazelbuild/bazel/issues/8900
-        "--incompatible_display_source_file_location",
-        'filter(".*\\.(h|cpp|cc|c|cxx)$", kind("source file", deps(//...)))',
-    ],
-    stderr=subprocess.DEVNULL,
-    universal_newlines=True,
-)
-source_files = [
-    Path(line.split(":")[0]) for line in source_files_query.splitlines()
-]
-
-# Filter into the Carbon source files that we'll find directly in the
-# workspace, and LLVM source files that need to be mapped through the merged
-# LLVM tree in Bazel's execution root.
-carbon_files = [
-    f.relative_to(directory)
-    for f in source_files
-    if f.parts[: len(directory.parts)] == directory.parts
-]
-llvm_files = [
-    Path("bazel-execroot/external").joinpath(
-        *f.parts[f.parts.index("llvm-project") :]
+
+def _build_generated_files(bazel: str) -> None:
+    print("Building the generated files so that tools can find them...")
+
+    # Now collect the generated file labels. Include some rules which generate
+    # files but aren't classified as "generated file".
```

Comment:
Tiny comment tweak given new structure.

```suggestion
    # Collect the generated file labels. Include some rules which generate
    # files but aren't classified as "generated file".
```